.github/workflows/build-deploy-dev.yml

```diff
@@ -5,42 +5,133 @@ on:
     types: [opened, reopened, synchronize, closed]
     branches:
       - main
+  workflow_dispatch:
+    inputs:
+      bumb:
+        description: "The type of semantic version increment to make. By default it is 'prerelease'."
+        required: false
+        default: "prerelease"
+        type: choice
+        options:
+          - major
+          - premajor
+          - minor
+          - preminor
+          - patch
+          - prepatch
+          - prerelease
+      preId:
+        description: "Identifier to be used to prefix version increments. By default it is 'prerelease' for pull request event and workflow dispatch will follow the last release tag prefix, if any."
+        required: false
+        type: choice
+        options:
+          -
+          - premajor
+          - preminor
+          - prepatch
+          - prerelease
+      releaseType:
+        description: "Select the release type. By default only tag will be created. if you want to create a release with the tag select the type."
+        type: choice
+        required: false
+        options:
+          -
+          - draft
+          - publish
+
 
 jobs:
   # Print variables for logging and debugging purposes
   checkEnv:
     name: Check Env variables
     environment: DEV
     runs-on: ubuntu-latest
-    if: ${{ github.event.pull_request.merged == true}}
+    if: github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch'
     steps:
       - name: Print Env Vars
         run: |
           echo Git Base Ref: ${{ github.base_ref }}
-          echo Git Build ID: ${{ github.event.number }}
           echo Git Pull Request Ref: ${{ github.event.pull_request.head.sha }}
           echo OC CLI Version: $(oc version)
 
+  # Create new tag.
+  createTag:
+    name: "Create tag"
+    if:  github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch' 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get bumb value
+        uses: haya14busa/action-cond@v1
+        id: getBumb
+        with:
+          cond: ${{ github.event_name == 'workflow_dispatch'}}
+          if_true: ${{ inputs.bumb }}
+          if_false: 'prerelease'
+
+      - name: Get bumb value, if provided
+        uses: haya14busa/action-cond@v1
+        id: getPreId
+        with:
+          cond: ${{ github.event_name == 'workflow_dispatch'}}
+          if_true: ${{ inputs.preId }}
+          if_false: 'prerelease'
+      - run: |
+          echo "bumb: ${{ steps.getBumb.outputs.value }}"
+          echo "prefix: ${{ steps.getPreId.outputs.value }}"
```

Comment:
Why is it called prefix/getPreId. Can we use one or another?

Reply:
preid was in variable name in the action, that's why I used the same. any suggestion?

Reply:
Maybe adapt the message if it makes sense `echo "PreId: ${{ steps.getPreId.outputs.value }}"`